ipn/ipnlocal/cert.go

```diff
@@ -122,10 +125,80 @@ func shouldStartDomainRenewal(dir, domain string, future time.Time) bool {
 		return false
 	}
 	lastRenewCheck[domain] = now
-	_, ok := getCertPEMCached(dir, domain, future)
+	_, ok := b.getCertPEMCached(dir, domain, future)
 	return !ok
 }
 
+// certStore provides a way to perist and retrieve TLS certificates.
+// As of 2023-02-01, we use store certs in directories on disk everywhere
+// except on Kubernetes, where we use the state store.
+type certStore interface {
+	Read(domain string, now time.Time) (*TLSCertKeyPair, error)
```

Comment:
is the error anything special? any particular types? is it an error if now is greater than the cert's expiry? does this auto-fetch a new one?

looks like caller ignores it entirely and just treated it as a boolean?

Reply:
shouldn't we at least define what the error for "Not found" is to prevent log spam later? currently the two impls have two different errors.

Reply:
made the errors be consistent, and plumbed it through the caller.